gen.go

```diff
@@ -1305,6 +1376,10 @@ func emptyValForField(f Field) (string, error) {
 	}
 }
 
+type Marshaler interface {
+	MarshalCBOR() ([]byte, error)
```

Comment:
Any reason to introduce a new interface instead of using the current `MarshalCBOR(w io.Writer) error` one?